### PR TITLE
feat(behavior_path_planner): handle overlapped road turn signal

### PR DESCRIPTION
**Pull request by purewater0901**
## Description
In this PR, I changed turn signal decider to enable it to handle overlapped trajectories. Results can be seen from the following video. 

Before this PR, the turn signal decider cannot handle overlapped trajectories. So it can activate wrong signals or deactivate signals. So this means, if a vehicle try to pass the same place with different angles, it cannot handle turn signals correctly. These problems can be solved with this PR.

- Before this PR
https://user-images.githubusercontent.com/43805014/197735625-69121eaa-c53b-4696-ad1d-24d29e430b04.mp4

- After this PR
https://user-images.githubusercontent.com/43805014/197686197-302df1d6-8da9-4ee7-a4c0-14d74130b920.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by mitsudome-r** [comment]
@purewater0901 Could you add more explain on what you are trying to solve?
I don't really get what you mean by "duplicate" trajectory, and it would be also nice to explain how it was behaving before this PR is merged.

**Comment by purewater0901** [comment]
@mitsudome-r Thank you for your comment. I updated PR title and description. In addition, I upload a video, which shows the result without this PR

File: planning/behavior_path_planner/src/turn_signal_decider.cpp [hunk 1]
```diff
@@ -28,31 +29,43 @@
...
+  const std::shared_ptr<const PlannerData> & planner_data, const PathWithLaneId & path,
   const TurnSignalInfo & turn_signal_info)
 {
   // Guard
   if (path.points.empty()) {
     return turn_signal_info.turn_signal;
   }
 
+  // Data
+  const double nearest_dist_threshold = 50.0;
```
Comment (takayuki5168) [inline planning/behavior_path_planner/src/turn_signal_decider.cpp:41]: Why using constant value.

File: planning/behavior_path_planner/src/turn_signal_decider.cpp [hunk 2]
```diff
@@ -148,10 +181,11 @@ boost::optional<TurnSignalInfo> TurnSignalDecider::getIntersectionTurnSignalInfo
 
     const auto & turn_signal_info = signal_queue.front();
     const auto & required_end_point = turn_signal_info.required_end_point;
-    const size_t nearest_seg_idx =
-      motion_utils::findNearestSegmentIndex(path.points, required_end_point);
+    const size_t nearest_seg_idx = motion_utils::findFirstNearestSegmentIndexWithSoftConstraints(
+      path.points, required_end_point, nearest_dist_threshold, nearest_yaw_threshold);
```
Comment (takayuki5168) [inline planning/behavior_path_planner/src/turn_signal_decider.cpp:185]: Is it okay to use find"First"NearestSegmentIndexWithSoftConstraints?
This function is designed to find "ego"'s nearest segment index.

File: planning/behavior_path_planner/src/turn_signal_decider.cpp (hunk 1 above)
Comment (purewater0901) [reply planning/behavior_path_planner/src/turn_signal_decider.cpp]: done

**Review by takayuki5168 — APPROVED** [review]
LGTM

**Pull request merged**